File: backend/src/main/java/com/dev/backend/service/CidadeService.java
package com.dev.backend.service;

import java.util.Date;
import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.dev.backend.entity.Cidade;
import com.dev.backend.repository.CidadeRepository;

@Service
public class CidadeService {

    @Autowired
    private CidadeRepository cidadeRepository;

    // INSERIR
    public Cidade inserir(Cidade cidade) {
        cidade.setDataCriacao(new Date());
        Cidade cidadeNova = cidadeRepository.saveAndFlush(cidade);
        return cidadeNova;
    }

    // ALTERAR
    public Cidade alterar(Cidade cidade) {
        cidade.setDataAtualizacao(new Date());
        return cidadeRepository.saveAndFlush(cidade);
    }

    // EXCLUIR
    public void excluir(Long id) {
        Cidade cidade = cidadeRepository.findById(id).get();
        cidadeRepository.delete(cidade);
    }

    // BUSCAR TODOS
    public List<Cidade> buscarTodos() {
        return cidadeRepository.findAll();
    }
}

File: sakai-react/src/service/cadastros/CidadeService.js
import axios from 'axios';

export class CidadeService {

    url = process.env.REACT_APP_URL_API;

    listarTodos(){
        return axios.get(this.url+'/cidade/');
    }

    inserir(cidade){
        return axios.post(this.url+'/cidade/', cidade);
    }

    alterar(cidade){
        return axios.put(this.url+'/cidade/', cidade);
    }

    excluir(id){
        return axios.delete(this.url+'/cidade/'+id);
    }
}